### DISCUSSION TITLE
Add download Supabase AI conversations button

### DISCUSSION THREAD

**Discussion by jacob-8** [supabase/supabase#18673] [Feature Requests]
Hey, can I contribute a simple button to download my Supabase AI conversations as a markdown file? I have a great time learning from your docs via AI and then it's a bit annoying to have to select a whole conversation and paste it into markdown. The select is error-prone for long conversations, and it does not preserve code block markings.

Lookin at you @gregnr

**Comment by gregnr** [comment]
Definitely open to this @jacob-8, good suggestion. Something that might be a bit simpler could be a copy-to-clipboard button that does the same thing. Would this work for you use case?

By the way, we've gotten lots of feedback about losing conversation history - there are plans to persist this in some way so that you don't lose your history every time you close the cmd+k menu.

**Reply by jacob-8** [reply]
Yes, a copy button would be plenty as a stop gap.

And yes, please do add history. That's really what I need. For now I can just copy and maintain history through manually creating notes files. If it's not too much to ask, adding semantic search to your history would make it next level. ChatGPT's history is almost useless for older than a couple weeks if you use the site a lot. So you would be dogfooding and pulling off a best in class feature.
